pkg/kubelet/logs/container_log_manager.go

```diff
@@ -470,3 +481,82 @@ func (c *containerLogManager) rotateLatestLog(ctx context.Context, id, log strin
 	}
 	return nil
 }
+
+// splitAndRotateLatestLog splits a log file into multiple smaller files if it exceeds MaxSize
+func (c *containerLogManager) splitAndRotateLatestLog(log string) ([]string, error) {
+	// Open the log file for reading
+	r, err := c.osInterface.Open(log)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open log %q: %v", log, err)
+	}
+	defer r.Close()
+
+	var parts []string
+	var partIndex int
+	var currentPart []byte
+	buffer := make([]byte, 1024) // Read in chunks of 1 KB
+
+	for {
+		n, err := r.Read(buffer)
```

Comment:
As this is a log file, would it make sense to read it line by line? Otherwise last line will be written only partly in most cases.